### ISSUE TITLE
What module should be responsible for persistence rules?

### ISSUE THREAD

**Issue by falvojr** [nglauber/books_jetpack#4]
Glauber, first congratulations on your exceptional work!
I am using this project as a basis for the development of an App and I would like to take a conceptual question with you, if possible:

I structured the data layers: `data_local` (Room) and `data_remote` (Retrofit, consuming a REST API). However, I would like to know what layer would be responsible for any rules related to the source of data persistence (local and remote)?

In this sense, your project currently configures a single data source: local (Room) or remote (Firebase). Conceptually I imagine that the `domain` layer could be responsible for persistence rules using multiple repositories, but in some references (like [this](https://github.com/android/architecture-samples)) I see that it is common to have a concrete implementation of the Repository pattern in the "data" layer.

Could you give your opinion, please?

**Comment by nglauber** [comment]
hey @falvojr. I'm glad this sample has been useful for you. 
Regarding your question, I think the solution would vary in according to the case. But in general, you will always have a single repository, but you can have two (or more) data sources. The repository will perform the required orchestrations between those data sources.
The local and remote data sources, as you mention, is a classic example. The repository can save/load the data locally and/or remotely, but from the domain layer standpoint, nothing changes, the use case will continue accessing a single repository as usual. 
It would be something like this sample [here](https://github.com/hitherejoe/GithubTrending/blob/task/mobile-ui/Data/src/main/java/co/joebirch/data/ProjectsDataRepository.kt). Here, every time the data are loaded (using `getProjects()`) these data are also saved locally.
To summarize, you will create a "master repository" which will be an implementation of repository interface which is composed by a local repository and a remote repository. The way of these data will be saved/retrieved is up to you 😅

**Comment by falvojr** [comment]
Perfect @nglauber ! Thank you very much for the quick response, it helped me a lot!
